docs/configuration/backends/kubernetes.md

```diff
@@ -159,7 +159,8 @@ The following general annotations are applicable on the Ingress object:
 | `traefik.ingress.kubernetes.io/whitelist-source-range: "1.2.3.0/24, fe80::/16"` | A comma-separated list of IP ranges permitted for access (6).                                                     |
 | `ingress.kubernetes.io/whitelist-x-forwarded-for: "true"`                       | Use `X-Forwarded-For` header as valid source of IP for the white list.                                            |
 | `traefik.ingress.kubernetes.io/app-root: "/index.html"`                         | Redirects all requests for `/` to the defined path. (4)                                                           |
-| `traefik.ingress.kubernetes.io/service-weights: <YML>`                          | Set ingress backend weights specified as percentage or decimal numbers in YAML. (5)                               |
+| `traefik.ingress.kubernetes.io/service-weights: <YML>`                          | Set ingress backend weights specified as percentage or decimal numbers in YAML. (5)            
+| `ingress.kubernetes.io/protocol: <NAME>`                                        | Set the protocol Traefik will use to communicate with pods.
```

Comment:
i thought ingress model is only for http/https protocol. this annotation allows users to set their own arbitrary protocol and once someone abuse it might cause some potential SQL-injection-like danger?🧐

Reply:
@yue9944882 The issue here is that h2c runs over http, but uses its own protocol. All of the other providers have the `traefik.protol` label, but the k8s provider does not. This allows users to specify h2c as a backend protocol so h2c will work as is does with other providers. Although your point about potential abuse is valid.

@SantoDE what do you think about instead of blindly assigning the value to `protocol`, running it through a select/switch with acceptable protocols? `http`/`https`/`h2c` so that someone can't add `http://myinjectionsite.com/endpoint.php http` as the protocol ;)